en/docs/debugging/debug.md

```diff
@@ -0,0 +1,258 @@
+# Debug Ballerina Programs
+
+When writing large-scale applications with complex logic, it is quite possible to have syntax, semantic,
+or runtime issues in your code. The Ballerina compiler comes in handy when detecting syntax and semantic issues.
+However, it is impossible for a compiler to detect runtime errors like logical errors, because they occur during the
+program execution after a successful compilation. This is where the dedicated debugging tooling support becomes important.
+
+Currently, we provide a feature-rich debugging experience for troubleshooting Ballerina applications
+via the <a href="https://marketplace.visualstudio.com/items?itemName=WSO2.Ballerina" target="_blank">Ballerina Visual Studio Code extension</a>.
+
+## Start a debug session
+
+The Ballerina extension provides multiple options to debug Ballerina applications, and the most convenient way will be using the context-aware debug codelens support.
+
+However, if you require launching debug sessions with additional configurations (i.e., program arguments, environment variables), you can use `launch.json` configurations
+(For more information, see [Start a debug session with configurations](#start-a-debug-session-with-configurations)).
+
+### Start a program debug session
```

Comment:
IMO, we need a short explanation about `program`, `test`, and `remote` debugging.

See this example - https://learn.microsoft.com/en-us/visualstudio/debugger/remote-debugging?view=vs-2022

@praneesha

Reply:
Will fix this from another PR.